src/material/schematics/ng-generate/m3-theme/index.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Rule, SchematicContext, Tree} from '@angular-devkit/schematics';
+import {Schema} from './schema';
+import {
+  argbFromHex,
+  themeFromSourceColor,
+  hexFromArgb,
+  Theme,
+} from '@material/material-color-utilities';
+
+let HUE_TONES = [0, 10, 20, 25, 30, 35, 40, 50, 60, 70, 80, 90, 95, 98, 99, 100];
+let NEUTRAL_HUE_TONES = HUE_TONES.concat([4, 6, 12, 17, 22, 24, 87, 92, 94, 96]);
+
+function getMaterialTheme(color: string): Theme {
+  try {
+    let argbColor = argbFromHex(color);
+    return themeFromSourceColor(argbColor, [
+      {
+        name: 'm3-theme',
+        value: argbColor,
+        blend: true,
+      },
+    ]);
+  } catch (e) {
+    throw new Error(
+      'Cannot parse the specified color ' +
+        color +
+        '. Please verify it is a hex color (ex. #ffffff or ffffff).',
+    );
+  }
+}
+
+function getColorPalettes(color: string): Map<string, Map<number, string>> {
+  const materialTheme = getMaterialTheme(color);
+  const palettes: Map<string, Map<number, string>> = new Map();
+  for (const [key, palette] of Object.entries(materialTheme.palettes)) {
+    const paletteKey = key.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
+    const tones = paletteKey === 'neutral' ? NEUTRAL_HUE_TONES : HUE_TONES;
+    const colorPalette: Map<number, string> = new Map();
+    for (const tone of tones) {
+      const color = hexFromArgb(palette.tone(tone));
+      colorPalette.set(tone, color);
+    }
+    palettes.set(paletteKey, colorPalette);
+  }
+  return palettes;
+}
+
+function getColorPalettesSCSS(colorPalettes: Map<string, Map<number, string>>): string {
+  let scss = '(\n';
+  for (const [variant, palette] of colorPalettes!.entries()) {
+    scss += '  ' + variant + ': (\n';
+    for (const [key, value] of palette.entries()) {
+      scss += '    ' + key + ': ' + value + ',\n';
+    }
+    scss += '  ),\n';
+  }
+  scss += ');';
+  return scss;
+}
+
+export function generateSCSSTheme(
+  colorPalettes: Map<string, Map<number, string>>,
+  themeTypes: string[],
+): string {
+  let scss = [
```

Comment:
Does the final output apply the theme to any components? Might be nice to include a few as an example, e.g. button and card

Reply:
I added in the documentation the way I assume people could import the file. Although it does seem like a good thing to directly replace the applied themes once we find the project's theme file. Let me know which is preferred here.